Apps/UnitTests/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+if((NOT WIN32) OR WINDOWS_STORE)
+    message(FATAL_ERROR "Unsupported platform: ${CMAKE_SYSTEM_NAME}")
+endif()
+
+include(FetchContent)
+FetchContent_Declare(
+  googletest
+  # Specify the commit you depend on and update it regularly.
+  URL https://github.com/google/googletest/archive/23ef29555ef4789f555f1ba8c51b4c52975f0907.zip
+)
+# For Windows: Prevent overriding the parent project's compiler/linker settings
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(googletest)
```

Comment:
Where does the code end up using this method?

Reply:
It seems to build fine on Windows without it. Here is the explanation for these lines: https://github.com/google/googletest/blob/master/googletest/README.md#visual-studio-dynamic-vs-static-runtimes

Doesn't seem to be necessary for us, not sure I understand why though.

Reply:
I mean where on disk does the FetchContent unzip the file?

Reply:
Oh, it looks like the googletest source code is being placed in a `_deps` subdirectory, I'll see if I can get it to unzip to `Dependencies\GoogleTest` instead

Reply:
I would recommend putting the downloaded files in the build folder somewhere as to not pollute the source folders.

Reply:
If we do end up putting it in Dependencies, then we need to put this folder in `.gitginore`.

Reply:
To be clear, by default, this ends up in the <build folder>/_deps.